### PR TITLE
Add job service syncing with Firebase every 12hrs

### PR DESCRIPTION
**Pull request by miltomasz**
Changes introduce job service triggering fetch action for Firebase twice a day. The frequency can be changed later on if needed.

### CODE REVIEW

File: app/src/main/java/pl/example/android/garbageapp/data/SectorTermRepository.java [hunk 1]
```diff
@@ -61,13 +63,30 @@ public synchronized static SectorTermRepository getInstance(SectorTermDao sector
...
+        });
     }
 
     private void startFetchSectorTerms() {
         mSectorTermsNetworkDataSource.startSectorTermsSyncService();
     }
 
+    private boolean isFetchNeeded() {
+        Date today = SectorTermsDateUtils.getNormalizedUtcDateForToday();
+        int futureWeather = mSectorTermDao.countAllFutureSectorTerms(today);
```
Comment (gbielanski) [inline app/src/main/java/pl/example/android/garbageapp/data/SectorTermRepository.java]: It is not a weather :)

File: app/src/main/java/pl/example/android/garbageapp/data/network/SectorTermsNetworkDataSource.java [hunk 2]
```diff
@@ -26,8 +33,17 @@
 
     private static final String LOG_TAG = SectorTermsNetworkDataSource.class.getSimpleName();
 
+    public static final int NUM_DAYS = 14;
+
```
Comment (gbielanski) [inline app/src/main/java/pl/example/android/garbageapp/data/network/SectorTermsNetworkDataSource.java]: NUM_DAYS should be greater. I think it should be 2 mounths (60 days) at least.

File: app/src/main/java/pl/example/android/garbageapp/data/network/SectorTermsNetworkDataSource.java [hunk 3]
```diff
@@ -54,6 +70,26 @@ public static SectorTermsNetworkDataSource getInstance(Context context, AppExecu
         return sInstance;
     }
 
+    public void scheduleRecurringFetchSectorTermsSync() {
+        Driver driver = new GooglePlayDriver(mContext);
+        FirebaseJobDispatcher dispatcher = new FirebaseJobDispatcher(driver);
+
+        Job syncSunshineJob = dispatcher.newJobBuilder()
```
Comment (gbielanski) [inline app/src/main/java/pl/example/android/garbageapp/data/network/SectorTermsNetworkDataSource.java]: It is not Sunshine

File: app/src/main/java/pl/example/android/garbageapp/data/SectorTermRepository.java [hunk 4]
```diff
@@ -61,13 +63,30 @@ public synchronized static SectorTermRepository getInstance(SectorTermDao sector
...
+        return futureWeather < SectorTermsNetworkDataSource.NUM_DAYS;
```
Comment (gbielanski) [inline app/src/main/java/pl/example/android/garbageapp/data/SectorTermRepository.java]: This condition is wrong. Dao countAllFutureSectorTerms will return number of terms. We cannot compare it to number of days.

**Review by gbielanski — CHANGES_REQUESTED** [review]

**Comment by miltomasz** [comment]
Hi,

I removed isFetchNeeded method as discussed. Now the app will be fetching sector terms in two ways:
- at the startup of the app
- next every 12hrs (twice a day). This will be handled by job service.

The issue to think about is how to set up a retry mechanism when app fails to download data (especially at startup).

**Review by gbielanski — APPROVED** [review]
Changes approved.

**Comment by gbielanski** [comment]
Let's leave discussion about sync fails and other special cases to consider in a separate topic.

**Pull request merged**